### PR TITLE
Coded features review: LP bond slashing & top-up

### LINKED ISSUE #731: Coded features review: LP bond slashing & top-up
The scenario tests LP bond slashing and top-up.

So far this has identified a core bug in the slashing part, once it's fixed we'll progress on writing the top-up tests. See progress of vegaprotocol/vega#3632 .

# ToDo:
Penalties for LP providers (aka bond slashing) is detailed in https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0044-lp-mechanics.md#penalties . Other relevant specs are https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0042-setting-fees-and-rewarding-lps.md but note that other aspects (LP orders, equity like share, distribution of fees) has been covered by other feature tests. 

We need to test the situation where the LP doesn't have enough balance across margin and general accounts and the margin account balance gets topped up from their LP bond account. 
- [ ] Test this happens as expected (bond account balance down, margin balance topped up)
- [ ] Moreover at this point a penalty is applied and the penalty amount should go to the insurance pool of the market.

### PR DESCRIPTION
**Pull request by davidsiska-vega**
The scenario tests LP bond slashing and top-up. 

So far this has identified a core bug in the slashing part, once it's fixed we'll progress on writing the top-up tests. See progress of https://github.com/vegaprotocol/vega/issues/3632 .

### CODE REVIEW

**Comment by cpuja84** [comment]
Hi @Jiajia-Cui - @davidsiska-vega asked me to assign this to you. Please feel free to catchup with him about the same. Thanks !

File: qa-scenarios/Debug-Featuretest-BondSlashing.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (davidsiska-vega) [inline qa-scenarios/Debug-Featuretest-BondSlashing.json]: This file shouldn't be included.

File: qa-scenarios/bond-slashing.feature [hunk 2]
```diff
@@ -1,27 +1,128 @@
...
+      | trader    | asset | market id | margin      | general    | bond     |
+      | lp1       | USD   | ETH/MAR22 | 33937710    | 2290       | 20000000 |
+      | trader1   | USD   | ETH/MAR22 | 33937710    | 290        | 20000000 |
+      | trader2   | USD   | ETH/MAR22 | 33937710    | 2290       | 20000000 |
 
-# If they have previously been slashed	
-# 	If there is a MTM move in their favour, bond account is topped up first
\ No newline at end of file
+
+    # And the order book should have the following volumes for market "ETH/MAR22":
```
Comment (davidsiska-vega) [inline qa-scenarios/bond-slashing.feature]: What's all this commented out stuff? If it should run, make it run, otherwise delete.

**Review by davidsiska-vega — COMMENTED** [review]
Please address the comments made and then we'll continue.

**Comment by davidsiska-vega** [comment]
Blocked by https://github.com/vegaprotocol/vega/issues/4627 .

**Review by edd — DISMISSED** [review]

**Review by Jiajia-Cui — APPROVED** [review]
LGTM

**Pull request merged**